pkg/trace/metrics/timing/timing.go

```diff
@@ -15,75 +15,102 @@ import (
 
 	"go.uber.org/atomic"
 
+	"github.com/DataDog/datadog-agent/pkg/trace/log"
 	"github.com/DataDog/datadog-agent/pkg/trace/metrics"
 )
 
-// AutoreportInterval specifies the interval at which the default set reports.
-const AutoreportInterval = 10 * time.Second
+// autoreportInterval specifies the interval at which the default set reports.
+var autoreportInterval = 10 * time.Second
 
 var (
-	defaultSet = NewSet()
-	stopReport = defaultSet.Autoreport(AutoreportInterval)
+	defaultSet *set
 )
 
 // Since records the duration for the given metric name as time passed since start.
 // It uses the default set which is reported at 10 second intervals.
-func Since(name string, start time.Time) { defaultSet.Since(name, start) }
+func Since(name string, start time.Time) {
+	m.Lock()
+	defer m.Unlock()
+	if defaultSet == nil {
+		log.Error("Timing hasn't been initialized, trace-agent metrics will be missing")
+		return
+	}
+	defaultSet.Since(name, start)
+}
+
+var m = sync.Mutex{}
```

Comment:
We should probably use an atomic here to avoid a double-lock (lock `m`, lock `defaultSet.mu`)